roles/federation-demo/tasks/deploy.yml

```diff
@@ -0,0 +1,75 @@
+---
+  - name: Check if project exists
+    stat:
+      path: "{{ CLONE_DEST }}"
+    register: project_stat
+
+  - name: Clean var/www
+    file:
+      state: absent
+      path: "{{ CLONE_DEST }}"
+    when: project_stat.stat.exists
+
+  - name: Clone the repository
+    shell: GIT_SSH_COMMAND="ssh -i {{ KEY_PATH }} -v -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null" {{ GIT_EXECUTABLE }} clone --single-branch --branch {{GIT_BRANCH}} {{ GIT_REPO }} {{ CLONE_DEST }}
```

Comment:
I would suggest you to use the **git** Ansible mode :
https://docs.ansible.com/ansible/2.9/modules/git_module.html